packages/0x.js/test/remaining_fillable_calculator_test.ts

```diff
@@ -0,0 +1,136 @@
+import 'mocha';
+import * as chai from 'chai';
+import BigNumber from 'bignumber.js';
+import { chaiSetup } from './utils/chai_setup';
+import { RemainingFillableCalculator } from '../src/order_watcher/remaining_fillable_calculator';
+import { SignedOrder, ECSignature } from '../src/types';
+import { TokenUtils } from './utils/token_utils';
+import { ZeroEx } from '../src/0x';
+
+chaiSetup.configure();
+const expect = chai.expect;
+
+describe('RemainingFillableCalculator', () => {
+    let calculator: RemainingFillableCalculator;
+    let signedOrder: SignedOrder;
+    let transferrableMakerTokenAmount: BigNumber;
+    let transferrableMakerFeeTokenAmount: BigNumber;
+    let remainingMakerTokenAmount: BigNumber;
+    let makerAmount: BigNumber;
+    let takerAmount: BigNumber;
+    let makerFee: BigNumber;
+    let isMakerTokenZRX: boolean;
+    const makerToken: string = '0x1';
+    const takerToken: string = '0x2';
+    const decimals: number = 4;
+    const zero: BigNumber = new BigNumber(0);
+    const zeroAddress = '0x0';
+    const signature: ECSignature = { v: 27, r: '', s: ''};
+    beforeEach(async () => {
+        [makerAmount, takerAmount, makerFee] = [ZeroEx.toBaseUnitAmount(new BigNumber(50), decimals),
+                                                ZeroEx.toBaseUnitAmount(new BigNumber(5), decimals),
+                                                ZeroEx.toBaseUnitAmount(new BigNumber(1), decimals)];
+        [transferrableMakerTokenAmount, transferrableMakerFeeTokenAmount] = [
+                                                ZeroEx.toBaseUnitAmount(new BigNumber(50), decimals),
+                                                ZeroEx.toBaseUnitAmount(new BigNumber(5), decimals)];
```

Comment:
All the above values are hard-coded and should never change. Why are they in the `beforeEach` block then?

Reply:
We are now changing these values in a few different test cases.